app/src/main/java/de/westnordost/streetcomplete/quests/construction/MarkCompletedHighwayConstruction.java

```diff
@@ -0,0 +1,119 @@
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment;
+
+public class MarkCompletedHighwayConstruction extends MarkCompletedConstruction
+{
+	@Inject
+	public MarkCompletedHighwayConstruction(OverpassMapDataDao overpassServer) {
+		super(overpassServer);
+	}
+
+	@Override public boolean download(BoundingBox bbox, MapDataWithGeometryHandler handler)
+	{
+		return overpassServer.getAndHandleQuota(getOverpassQuery(bbox), handler);
+	}
+
+	/** @return overpass query string to get streets marked as under construction but excluding ones
+	 * - with invalid construction tag
+	 * - with tagged opening date that is in future
+	 * - recently edited (includes adding/updating check_date tags)
+	 * - with fixme (as in some situations this quest may add fixme note as an edit)
+	 * . */
+	private static String getOverpassQuery(BoundingBox bbox)
+	{
+		ArrayList<String> acceptedConstructionValues = new ArrayList<String>(Arrays.asList(OsmTaggings.ALL_ROADS));
+		acceptedConstructionValues.add("cycleway");
+		acceptedConstructionValues.add("footway");
+		acceptedConstructionValues.add("path");
+
+		String groupName = ".roads_under_construction";
+		String resultsName = ".roads_for_review";
+		return OverpassQLUtil.getGlobalOverpassBBox(bbox) +
+			getQueryPart("highway", groupName, 14) +
+			"(" +
+			"way[construction]" +
+			"[construction !~ \"^("+ TextUtils.join("|", acceptedConstructionValues)+")$\"]" +
+			") -> .invalid_construction_type;" +
+			"(" +
+			"  way[highway=construction][fixme];" +
+			") -> .with_fixme;" +
+			"((" +
+			groupName + " - .invalid_construction_type)" +
+			"- .with_fixme" +
+			") -> " + resultsName + ";" +
+			resultsName + " out meta geom;";
+	}
+
+	public void applyAnswerTo(Bundle answer, StringMapChangesBuilder changes)
+	{
+		if(answer.getBoolean(YesNoQuestAnswerFragment.ANSWER)) {
+			if(isAnyAccessTagPresent(changes)){
+				// there are some access tag present that are likely to be temporary
+				// and valid only during construction - marking construction as finished
+				// would likely cause nonobvious tagging mistake
+				changes.add("fixme", "construction is completed but there are access tags that maybe were true only during construction period");
+				return;
+			}
+			String constructionValue = changes.getPreviousValue("construction");
+			if(constructionValue == null) {
+				constructionValue = "road";
+			}
+			changes.modify("highway", constructionValue);
+			changes.deleteIfPresent("construction");
+			changes.deleteIfPresent("opening_date");
+			changes.deleteIfPresent(OsmTaggings.SURVEY_MARK_KEY);
+		} else {
+			changes.addOrModify(OsmTaggings.SURVEY_MARK_KEY, DateHandler.getCurrentDateString());
+		}
+	}
+
+	private boolean isAnyAccessTagPresent(StringMapChangesBuilder changes){
+		for(String accessTag : OsmTaggings.POPULAR_ROAD_ACCESS_TAGS){
+			if(changes.getPreviousValue(accessTag) != null) {
+				return true;
+			}
+		}
+		return false;
+	}
```

Comment:
> why not any, or at least, any documented one

I though that including all defined at wiki would require either update treadmill (as people invent new ones and add them at https://wiki.openstreetmap.org/wiki/Key:access#Land-based_transportation ) or it would be weird "all common ones and all defined at February 2018.

But I may update list to include all listed at https://wiki.openstreetmap.org/wiki/Key:access#Land-based_transportation - should I do it?

>  And why not remove the access tag?

https://www.openstreetmap.org/way/83230134 has access tags of way that will be build, https://www.openstreetmap.org/way/540501408/history has something that I am unable to interpret and maybe should be kept maybe deleted https://www.openstreetmap.org/way/88702995 has tags that refer to what will be constructed - for more cases like this see http://overpass-turbo.eu/s/wGm

overall, in this case I prefer to avoid making edit that in 96% will be correct in 3% will be highly confusing and 1% will break something and irritate normal mappers (% obviously based on biased sample and guessing but even if silent damage, without mistake by editing person, would happen in 0.1% of cases I would still avoid edit like this)

Reply:
> I though that including all defined at wiki would require either update treadmill 

I mean, why not access=*?

> And why not remove the access tag?

Wouldn't then a simpler logic be feasible as well? If access=no|construction_vehicles or similar -> remove access. Anything else, don't change anything?

Reply:
See for example https://www.openstreetmap.org/way/83230134 where access=no should stay.

> I mean, why not access=*?

Just access key? And ignore other access keys? Note that it is quite common to set value also in other keys - foot is the most popular but there are also more complex cases.

Reply:
> Just access key? And ignore other access keys?

Ah, I understand.